Okta/OktaAuth/OktaAuthTask.swift

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+class OktaAuthTask<T> {
+    private(set) var config: OktaAuthConfig?
+ 
+    init(config: OktaAuthConfig?) {
```

Comment:
How about to pass OktaApi instance as DI? It is actively used by inherited classes